tests/test-sources/plugins/bufferlines/bufferline.nix

```diff
@@ -1,7 +1,29 @@
+{ lib, pkgs, ... }:
 {
-  empty = {
-    plugins.bufferline.enable = true;
-  };
+  empty =
+    { config, ... }:
+    {
+      plugins.bufferline.enable = true;
+
+      assertions = [
+        {
+          assertion =
+            config.extraPlugins != [ ]
+            && lib.any (
+              x: lib.trace "${x.pname or ""}" x.pname or null == "bufferline.nvim"
```

Comment:
Looking on noogle, there's also [`lib.getName`](https://noogle.dev/f/lib/getName), although I think that could fail when encountering submodule-type plugin definitions...